src/pages/registration/references_tab/components/SearchContainerField.tsx

```diff
@@ -0,0 +1,107 @@
+import React, { useState } from 'react';
+import { Field, FieldProps, getIn, useFormikContext } from 'formik';
+import { Typography } from '@material-ui/core';
+import { Autocomplete } from '@material-ui/lab';
+import Truncate from 'react-truncate';
+import { Registration } from '../../../../types/registration.types';
+import { autocompleteTranslationProps } from '../../../../themes/mainTheme';
+import useSearchRegistrations from '../../../../utils/hooks/useSearchRegistrations';
+import { AutocompleteTextField } from '../../../../components/AutocompleteTextField';
+import EmphasizeSubstring from '../../../../components/EmphasizeSubstring';
+import { StyledFlexColumn } from '../../../../components/styled/Wrappers';
+import { displayDate } from '../../../../utils/date-helpers';
+import useDebounce from '../../../../utils/hooks/useDebounce';
+import { API_URL } from '../../../../utils/constants';
+import { getRegistrationPath } from '../../../../utils/urlPaths';
+import { createSearchQuery } from '../../../../utils/searchHelpers';
+import { ReferenceFieldNames, RegistrationSubtype } from '../../../../types/publicationFieldNames';
+
+interface SearchContainerFieldProps {
+  fieldName: string;
+  searchSubtypes: RegistrationSubtype[];
+  label: string;
+  placeholder: string;
+}
+
+const SearchContainerField = (props: SearchContainerFieldProps) => {
+  const { values, setFieldValue, setFieldTouched } = useFormikContext<Registration>();
+  const [searchTerm, setSearchTerm] = useState('');
+  const debouncedSearchTerm = useDebounce(searchTerm);
+
+  const optionsSearchQuery = createSearchQuery({
+    searchTerm: debouncedSearchTerm,
+    properties: [{ key: ReferenceFieldNames.SUB_TYPE, value: props.searchSubtypes }],
+  });
+  const [containerOptions, isLoadingContainerOptions] = useSearchRegistrations(optionsSearchQuery);
```

Comment:
liker heller ikke `containerOptions`. Det sier meg veldig lite. synes `searchContainerOptions` er bedre, men sier ikke så veldig mye det heller.

Reply:
ja, blir litt jalla uansett den der. For corrigendum er det `journalArticleOptions`, for bokkapittel er det `bookAnthologyOptions`, osv, så det blir bare å finne noe som passer best mulig for alle. Kan godt bruke `searchContainerOptions`